single_tech_samples/synapseanalytics/sample3_integration_testing/infrastructure/modules/appinsights.bicep

```diff
@@ -0,0 +1,25 @@
+param project string
+@allowed([
+  'dev'
+  'stg'
+  'prod'
+])
+param env string
+param location string = resourceGroup().location
+param deployment_id string
+
+
+resource appinsights 'Microsoft.Insights/components@2020-02-02-preview' = {
+  name: '${project}-appi-${env}-${deployment_id}'
+  location: location
+  tags: {
+    DisplayName: 'Application Insights'
+    Environment: env
+  }
```

Comment:
Bubbling this value up to the main.bicep, and passing it down as a param to all the modules may allow you to tag all resources created by all modules in a consistent manner

Reply:
Even if values are unique per module, the main.bicept level would be appropriate so that it's easier to spot any inconsistent naming conventions amongst the modules. 

Example - I noticed that the keyvault module uses PascalCase, while application insight a different convention.

Reply:
I pulled the bicep and scripts from the e2e sample so there would be a similar feel in the single tech sample and the e2e sample. I am happy to make these changes to this section and I do agree this would be nice to have. However, since after merging this PR there would be duplicate code in the repo, I feel that if I change the code here, it should also be changed in the e2e sample or maybe there is another solution for creating a reusable module instead of copying the code around. Open to more thoughts for sure! cc @devlace - including you in the conversation and happy to hear your thoughts!